src/actions/measurements.js

```diff
@@ -102,7 +102,15 @@ export const loadMeasurements = (json) => (dispatch, getState) => {
     throw new Error("tree not loaded");
   }
 
-  const collections = json["collections"];
+  /**
```

Comment:
An alternative approach here would be to use a flag to indicate whether the (non-idempotent) transform has been done. This should save memory & be faster as moves back to a previously loaded measurements panel won't need to clone + retransform the data. Something like:

```diff
diff --git a/src/actions/measurements.js b/src/actions/measurements.js
index 6909c560..4740e2d7 100644
--- a/src/actions/measurements.js
+++ b/src/actions/measurements.js
@@ -107,6 +107,8 @@ export const loadMeasurements = (json) => (dispatch, getState) => {
     throw new Error("Measurements JSON does not have collections");
   }
 
+  if (json?.__collections_transformed !== true) {
+  // following forEach block would need to be indented. Could also be pulled out into a function - i.e. collections.forEach(transform).
   collections.forEach((collection) => {
     /**
      * Keep backwards compatibility with single value threshold.
@@ -192,6 +194,8 @@ export const loadMeasurements = (json) => (dispatch, getState) => {
       })
     );
   });
+  json.__collections_transformed = true;
+  }
 
   // Get the collection to display to set up default controls
   // TODO: consider url query parameter?
```

Reply:
(Maybe use a [symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) instead of a string key)

Reply:
Thank you for the thoughts on performance here!

I went down a slightly different route to combine the fetch and transformation so that the transformation only happens on the initial fetch of the JSON, see 338db0aa28f61252808aeda25d45d2ef42ecb1cc.

Reply:
Hmm, just realized this method breaks the loading of the measurements sidecar JSON in auspice.us since it handles the dropped files instead of fetching from server.

Reply:
Refactored in 69e1d9c9466897441139a6d23e512b34955907a3 to keep the parsing in a separate function so that it can be called in auspice.us `handleDroppedFiles` to parse the dropped measurements sidecar JSON.

Reply:
See updates for auspice.us in https://github.com/nextstrain/auspice.us/pull/57